pkg/api/mock/matcher/example_test.go

```diff
@@ -0,0 +1,92 @@
+package matcher
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.anx.io/go-anxcloud/pkg/api/mock"
+)
+
+// Object is a custom gomega matcher which is used to test for matching objects
+// on the MockAPI instance by comparing the type and provided field names
+func ExampleObject() {
+	defer GinkgoRecover()
+
+	a := mock.NewMockAPI()
+
+	if err := a.Create(context.TODO(), &clouddnsv1.Record{ZoneName: "example.com", Name: "webmail"}); err != nil {
+		log.Fatalf("failed to create DNS record: %s", err)
+	}
+
+	ok := Expect(a.Existing()).To(
+		ContainElement(
+			Object(&clouddnsv1.Record{ZoneName: "example.com"}, "ZoneName"),
+		),
+	)
+
+	fmt.Println(ok)
+
+	// Output: true
+}
+
+// Destroyed is a custom gomega matcher which is used to test
+// if an object was destroyed
+func ExampleDestroyed() {
+	defer GinkgoRecover()
+
+	a := mock.NewMockAPI()
+
+	record := &clouddnsv1.Record{ZoneName: "example.com", Name: "webmail"}
+
+	if err := a.Create(context.TODO(), record); err != nil {
+		log.Fatalf("failed to create record: %s", err)
+	}
+
+	if err := a.Destroy(context.TODO(), &clouddnsv1.Record{Identifier: record.Identifier}); err != nil {
+		log.Fatalf("failed destroy record: %s", err)
+	}
+
+	ok := Expect(a.All()).To(
+		ContainElement(
+			Destroyed(1),
```

Comment:
please combine this with a check to make sure our Record was deleted, as this would be the most common usage of this